### DISCUSSION TITLE
Is it possible to tell if my function is running in a reactive context?

### DISCUSSION THREAD

**Discussion by tomasreimers** [mobxjs/mobx#3862] [Q&A]
I'm building a library on-top of mobx. I want to throw an error if a specific function is called outside of a reactive context (specifically, I expect it to be called from within a react component and want to make sure the observer HOC wasn't forgotten). 

Is that possible?

**Comment by kubk** [comment]
You may find it useful: https://mobx.js.org/configuration.html#observablerequiresreaction-boolean

**Comment by urugator** [comment]
`mobx._getGlobalState().trackingContext == null`
https://github.com/mobxjs/mobx/blob/13a74fe43de683249b2fd17145957b78f5befee4/packages/mobx/src/core/globalstate.ts#L49

`mobx._getGlobalState().trackingContext?.toString().endsWith('.render()]')` inside class component
`mobx._getGlobalState().trackingContext?.toString().startsWith('Reaction[observer')` inside func component
